crypto/asn1/tasn_utl.c

```diff
@@ -117,8 +117,10 @@ int asn1_do_lock(ASN1_VALUE **pval, int op, const ASN1_ITEM *it)
     if (op == 0) {
         *lck = 1;
         *lock = CRYPTO_THREAD_lock_new();
-        if (*lock == NULL)
+        if (*lock == NULL) {
+            ASN1err(ASN1_F_ASN1_DO_LOCK, ERR_R_MALLOC_FAILURE);
```

Comment:
I'm still going to argue that an individual function should work *consistently*, so for this one, there should be an error code for *each* `return 0` or none at all.